dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/inbound/NettyTCPWriteRequestContext.java

```diff
@@ -186,97 +200,127 @@ public void setBuffer(WsByteBuffer buf) {
         }
 
     }
+    
+    public void queuePrefixObject(Object object) {
+        this.prefixQueue.add(object);
+    }
+
+    private void awaitChannelFuture(ChannelPromise future, String failureMsg)
+        throws IOException, InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        future.addListener(f -> latch.countDown());
+        latch.await();
+        if (!future.isSuccess()) {
+            throw new IOException(failureMsg, future.cause());
+        }
+    }
+
+    private void verifyTimeout(int timeout) {
+        // Verify timeout and add it to the timeout handler
+        long timeoutNanos = TimeUnit.MILLISECONDS.toNanos(timeout);
+        WriteTimeoutHandler timeoutHandler = nettyChannel.pipeline().get(WriteTimeoutHandler.class);
+        if (Objects.isNull(timeoutHandler)) {
+            nettyChannel.pipeline().addLast("writeTimeoutHandler", new WriteTimeoutHandler(timeout, TimeUnit.MILLISECONDS));
+        } else if(timeoutHandler.getTimeout() != timeoutNanos) {
+            // Updated timeout so need to do so here as well
+            timeoutHandler.setTimeout(timeout, TimeUnit.MILLISECONDS);
+        }
+    }
 
     @Override
     public long write(long numBytes, int timeout) throws IOException {
-        AtomicLong writtenBytes = new AtomicLong(0);
-        // Check if "Content-Length" is set for this channel
-        boolean hasContentLength = nettyChannel.hasAttr(NettyHttpConstants.CONTENT_LENGTH) && Objects.nonNull(nettyChannel.attr(NettyHttpConstants.CONTENT_LENGTH).get());
+        if (nettyChannel.eventLoop().inEventLoop()) {
 
-        //check if wsoc
+            throw new IllegalStateException("Cannot invoke a blocking write on the Netty event loop thread.");
```

Comment:
Add to translations issue, if not already there

Reply:
Added to issue https://github.com/OpenLiberty/open-liberty/issues/31761